packages/sfu/server.ts

```diff
@@ -0,0 +1,121 @@
+import config from './config'
+import express from 'express'
+import fs from 'fs'
+import url from 'url'
+import protoo from 'protoo-server'
+import {types as mediasoupTypes, createWorker} from 'mediasoup'
+import Room from './lib/Room'
+import {AwaitQueue} from 'awaitqueue'
+import getVerifiedAuthToken from 'parabol-server/utils/getVerifiedAuthToken'
+import {isAuthenticated, isTeamMember} from 'parabol-server/utils/authorization'
+import checkBlacklistJWT from 'parabol-server/utils/checkBlacklistJWT'
+import Logger from './lib/Logger'
+
+const logger = new Logger()
+const mediasoupWorkers = []
+const rooms = new Map<string, Room>()
+const queue = new AwaitQueue()
+
+async function getOrCreateRoom(roomId: string) {
+  if (!rooms.get(roomId)) {
+    const worker = getMediaSoupWorker()
+    const room = await Room.create(roomId, worker)
+    rooms.set(roomId, room)
```

Comment:
there exists a race condition here.
server A & B both call `rooms.get`, it returns false for both
server A & B both create the room & call `rooms.set`
whichever returns last (e.g. B) would then be the only one that exists in the the `rooms` map.

Reply:
@mattkrick did you see where the function is called from? it gets put onto an in-memory queue, where each task is processed sequentially one by one.